store/tikv/region_cache.go

```diff
@@ -26,13 +27,24 @@ import (
 	goctx "golang.org/x/net/context"
 )
 
+const (
+	rcDefaultRegionCacheTTL = time.Minute * 2
+)
+
 // RegionCache caches Regions loaded from PD.
 type RegionCache struct {
 	pdClient pd.Client
-	mu       struct {
+	/*
+	 * this lock is used to read/write the map of ttls
+	 * within the critical section protected by this lock,
+	 * mu.lock should not be invoked.
+	 */
+	ttlLock sync.RWMutex
+	mu      struct {
 		sync.RWMutex
 		regions map[RegionVerID]*Region
 		sorted  *llrb.LLRB
+		ttls    map[RegionVerID]time.Time
```

Comment:
How about move the `ttl` field to `Region`, so no extra map need to be maintained.

Reply:
Region should not have a field ttl.
If we want to save a map structure, we can merge the two maps as
regions map[RegionVerID]<*Region, time.Time>.
However, this may incur updating a lot of codes.
So, I prefer keeping the two maps separately.

Reply:
@atmzhou 
Then how about `accessTime`, `Region` can have an accessTime field.

Reply:
@coocood @tiancaiamao 
I have make a structure named CachedRegion as follows:
type CachedRegion struct {
region Region
ttl        int64
}
Accordingly, each RegionCache will have a map map[RegionVerID]*CachedRegion.
In this way, we save a map structure.
I think this should conform to your thought.

Reply:
@atmzhou 
`ttl` is changing over time, we can't store a `ttl`value in field.